ADALiOS/ADALiOS/ADAuthenticationContext.m

```diff
@@ -121,24 +127,36 @@ - (id)init
 
 - (id)initWithAuthority:(NSString*) authority
       validateAuthority:(BOOL)bValidate
-        tokenCacheStore:(id<ADTokenCacheStoring>)tokenCache
+            sharedGroup:(NSString*)sharedGroup
                   error:(ADAuthenticationError* __autoreleasing *) error
 {
     API_ENTRY;
     NSString* extractedAuthority = [ADInstanceDiscovery canonicalizeAuthority:authority];
     RETURN_ON_INVALID_ARGUMENT(!extractedAuthority, authority, nil);
     
-    self = [super init];
-    if (self)
+    if (!(self = [super init]))
     {
-        _authority = extractedAuthority;
-        _validateAuthority = bValidate;
-        _tokenCacheStore = tokenCache;
-        _credentialsType = AD_CREDENTIALS_EMBEDDED;
+        return nil;
     }
+    
+    _authority = extractedAuthority;
+    _validateAuthority = bValidate;
+    _credentialsType = AD_CREDENTIALS_EMBEDDED;;
```

Comment:
consider renaming AD_CREDENTIAL_* to something meaningful? I like embedded vs Auto part.